### ISSUE TITLE
Source maps not loading in production

### ISSUE THREAD

**Issue by thavelick** [zeit/next.js#12176]
# Bug report

## Describe the bug

I am unable to get production source maps to work on a bare bones Next application. With significant tweaking, I can get the source map files to generate, but they don't show up in the browser. 

According to this, albeit old, blog entry: https://nextjs.org/blog/next-5, putting:

```
 webpack(config, { dev }) {
    if (!dev) {
      config.devtool = 'source-map'
    }
    return config
  }
```
should enable source maps in production, but it doesn't generate them at all in production builds.

Digging into the source of https://github.com/zeit/next-plugins/tree/master/packages/next-source-maps I determined that I can get production source maps to *generate* with the following configuration:

```
  webpack: (config) => {
    // enable source maps for webpack
    config.devtool = 'source-map';
    for (const plugin of config.optimization.minimizer) {
      if (plugin.constructor.name === 'TerserPlugin') {
        plugin.options.sourceMap = true;
      }
    }

    return config;
  }
```
but the source maps won't actually *load* in the browser. Interestingly, Chrome detects that a source map is present, but it doesn't actually appear to parse it. This is shown in this screen shot:

<img width="1421" alt="Chrome detecting source map but not parsing it" src="https://user-images.githubusercontent.com/1319124/80228227-0469e600-860c-11ea-9cc4-8dec1e123c65.png">

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Create a bare bones next app
2. make `next.config.js` contain:
```
module.exports = {
  webpack: (config) => {
    // enable source maps for webpack
    config.devtool = 'source-map';
    console.log('doing things')
    for (const plugin of config.optimization.minimizer) {
      if (plugin.constructor.name === 'TerserPlugin') {
        console.log('doing terseer things')
        plugin.options.sourceMap = true;
      }
    }

    return config;
  }
}
```

3. create `pages/index.js` contain something that is simple but would get obviously minified like:
```
export default function Index() {
  const name = "Tristan";
  return (
    <div>
    <p>Hello {name}</p>
    </div>
  );
}
```

4. Start the app in development mode with `npm run dev`
5. Open `http://localhost:3000` in Chrome (or another browser)
6. Open the developer console `Sources` tab and navigate to` _next/static/development/pages/index.js` and verify that the source map loads and the code is readable.
7. ctrl-c in your terminal to kill the dev app
8. Run `npm run build` to kick off a production build
9. Run `ls .next/static/` and note the random hash for this build
10. Run `ls .next/static/<build-hash>/pages/` and notice that an indext.js.map file does get generated
11. Run `npm run start`
12. Open `http://localhost:3000` in Chrome (or another browser)
13. Open the developer console `Sources` tab and navigate to` _next/static/<build-hash>/pages/index.js`
14. You'll see that the code is not readable, yet chrome will say "source map detected"

## Expected behavior

At the last step above, the code should be readable just like it is for the development build

## Screenshots

See above

## System information

- OS: verified on mac and linux builds
- Browser (if applies) prod source maps do not load on chrome, firefox or safari, but chrome does detect their existence
- Version of Next.js: 9.3.5
- Version of Node.js: v12.16.2 (also verified with v10.16.3)

**Comment by timneutkens** [comment]
Fixed by #13018

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
